src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx

```diff
@@ -1,183 +1,125 @@
-
-          const keypair = await getKeypair()
-
-          const txnOwnerSigned = await txn.sign({
-            owner: keypair,
-          })
-
-          const stakingServerSignedTxnStr = await Onboarding.getOnboardingSignedTransaction(
-            address,
-            txnOwnerSigned.toString(),
-          )
-
-          const stakingServerSignedTxn = AddGateway.txnFromString(
-            stakingServerSignedTxnStr,
-          )
-
-          await submitTxn(stakingServerSignedTxn.toString())
-        } catch (error) {
-          await handleError(error, 'add_gateway')
-          return
-        }
+        // Gateway QR scanned
+        updateParams.addGatewayTxn = qrAddGatewayTxn
       } else {
-        try {
-          const ownerAddress = await getAddress()
-          const ownerKeypairRaw = await getSodiumKeypair()
-
-          if (!ownerAddress?.b58 || !ownerKeypairRaw) {
-            await handleError(false, 'add_gateway')
-            return
-          }
-
-          const addGatewayResponse = await createGatewayTxn(
-            ownerAddress?.b58,
-            ownerKeypairRaw,
-          )
-          await submitTxn(addGatewayResponse)
-        } catch (error) {
-          await handleError(error, 'add_gateway')
-          return
-        }
+        // Gateway BLE scanned
+        const addGatewayTxn = await createGatewayTxn(ownerAddress)
+        updateParams.addGatewayTxn = addGatewayTxn
       }
     }
 
     // construct and publish assert location
     if (params.coords) {
       const [lng, lat] = params.coords
-      try {
-        const onboardingRecord = params?.onboardingRecord
-        const assertLocTxnResponse = await assertLocationTxn({
-          gateway: address,
-          lat,
-          lng,
-          decimalGain: params.gain,
-          elevation: params.elevation,
-          onboardingRecord,
-          dataOnly: false,
-        })
-        if (assertLocTxnResponse) {
-          await submitTxn(assertLocTxnResponse)
-          setFinished(true)
-          return
-        }
-        handleError(false, 'assert_location')
-      } catch (error) {
-        handleError(error, 'assert_location')
-      }
-    } else {
-      setFinished(true)
+      const onboardingRecord = params?.onboardingRecord
+
+      const assertLocationTxn = await Location.createLocationTxn({
+        gateway: hotspotAddress,
+        lat,
+        lng,
+        decimalGain: params.gain,
+        elevation: params.elevation,
+        dataOnly: false,
+        owner: ownerAddress,
+        // currentLocation: '', // If reasserting location, put previous location here
+        makerAddress: onboardingRecord.maker.address,
+        locationNonceLimit: onboardingRecord.maker.locationNonceLimit || 0,
+      })
+      updateParams.assertLocationTxn = assertLocationTxn.toString()
     }
+
+    const url = WalletLink.createUpdateHotspotUrl(updateParams)
+    if (!url || !Linking.canOpenURL(url)) return
```

Comment:
Should we show an alert or something?

Reply:
We could, but I guess I'd rather leave how it's handled up to the makers.